topologies/single.go

```diff
@@ -118,7 +118,7 @@ func (t *SingleTopology) GenerateBGPPeers(_ []corev1.Node, _ map[*corev1.Node]bo
 
 	bgpPeerConfigs = append(bgpPeerConfigs, *clientConfig)
 
-	return bgpPeerConfigs
+	return bgpPeerConfigs, nil
```

Comment:
I'm a bit cold to send back a nil as collecion. ususally it is a bad practice. The current code works well because `range` handkes nil, but make the codebase not so robust.

Reply:
Thank you! Returning an empty list now: []calicoApi.BGPPeer{}